pytorch_lightning/utilities/distributed.py

```diff
@@ -89,6 +89,12 @@ def gather_all_tensors(result: Union[torch.Tensor], group: Optional[Any] = None)
     if group is None:
         group = torch.distributed.group.WORLD
 
+    if not result.is_contiguous():
+        rank_zero_warn('Syncing with `gather_all` requires input to be contiguous'
+                       ' memory allocated. Will convert to contiguous format.',
+                       UserWarning)
+        result = result.contiguous()
```

Comment:
```suggestion
    result = result.contiguous()
```

The docs on `torch.Tensor.contiguous` state:
> If self tensor is already in the specified memory format, this function returns the self tensor. 

This means, that it should be a no-op if tensor is already contiguous and thus we don't need the check here.

Reply:
thanks, learned something new. Changed to your suggestion 👍